### PR TITLE
Added-cpu-core-and-memory-range-fot-type-offers-CustomConstrained

### PR DESCRIPTION
**Pull request by LayonRibeiro**
### Description

This PR intends to enable the visualization of the amount of CPU/memory allocated in custom constrained compute offerings, which is currently not possible, thus improving the user experience.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

<summary>  Before </summary>

before

<summary>  After </summary>

after

### How Has This Been Tested?
The CPU/memory ranges appeared in the offerings table under Compute Offerings.

### CODE REVIEW

File: ui/src/components/view/ListView.vue [hunk 1]
```diff
@@ -240,6 +240,12 @@
       <template v-if="column.key === 'agentstate'">
         <status :text="text ? text : ''" displayText />
       </template>
+      <template v-if="column.key === 'cpunumber'">
+        <span>{{ record.serviceofferingdetails ? `${record.serviceofferingdetails.mincpunumber} - ${record.serviceofferingdetails.maxcpunumber}` : record.cpunumber }}</span>
```
Comment (winterhazel) [inline ui/src/components/view/ListView.vue:244]:
```suggestion
        <span>{{ record.serviceofferingdetails?.mincpunumber && record.serviceofferingdetails?.maxcpunumber ? `${record.serviceofferingdetails.mincpunumber} - ${record.serviceofferingdetails.maxcpunumber}` : record.cpunumber }}</span>
```

This results in an error when `serviceofferingdetails` contains details other than `mincpunumber` and `maxcpunumber`, such as when the compute offering is associated to a zone. We should check if both `mincpunumber` and `maxcpunumber` are present instead.

image

File: ui/src/components/view/ListView.vue [hunk 2]
```diff
@@ -240,6 +240,12 @@
...
+      </template>
+      <template v-if="column.key === 'memory'">
+        <span>{{ record.serviceofferingdetails ? `${record.serviceofferingdetails.minmemory} - ${record.serviceofferingdetails.maxmemory}` : record.memory }}</span>
```
Comment (winterhazel) [inline ui/src/components/view/ListView.vue:247]:
```suggestion
        <span>{{ record.serviceofferingdetails?.minmemory && record.serviceofferingdetails?.maxmemory ? `${record.serviceofferingdetails.minmemory} - ${record.serviceofferingdetails.maxmemory}` : record.memory }}</span>
```

Same situation here

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Review by JoaoJandre — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@JoaoJandre @DaanHoogland some PRs targeting 4.20.1 are merged on main - should they go to 4.20 branch?

**Comment by JoaoJandre** [comment]
@rohityadavcloud Considering that they were not targeting 4.19 (a maintenance branch), I assumed the authors wanted their changes to be made on the main branch directly. I'll fix the milestone. If the authors want their changes to be on the 4.20 branch, I volunteer to backport the PRs (#9930, #9708, #9956).

**Pull request merged**